backends/cuda/ceed-cuda-vec.c

```diff
@@ -88,10 +89,11 @@ static int CeedVectorSetArrayDevice_Cuda(const CeedVector vec,
 
   switch (cmode) {
   case CEED_COPY_VALUES:
-    ierr = cudaMalloc((void **)&data->d_array_allocated, bytes(vec));
-    CeedChk_Cu(ceed, ierr);
-    data->d_array = data->d_array_allocated;
-
+    if (!data->d_array_allocated) {
```

Comment:
Same as previously, I think you want to check:
```c++
if (!data->d_array)
```

Reply:
Good point